File: leetcode/csharp/1545.find-kth-bit-in-nth-binary-string.cs
public class Solution {
    public char FindKthBit(int n, int k) {
        if (n == 1) {
            return '0';
        }
        int midIndex = 1 << (n-1);
        if (midIndex == k)  {
            return '1';
        }
        if (k < midIndex) {
            return FindKthBit(n-1, k);
        }
        if (FindKthBit(n-1, 2 * midIndex - k) == '0') {
            return '1';
        }
        return '0';
    }
}

File: leetcode/rust-leetcode/src/1545.find-kth-bit-in-nth-binary-string.rs
impl Solution {
    pub fn find_kth_bit(n: i32, k: i32) -> char {
        if n == 1 {
            return '0';
        }
        let midIndex = 2_i32.pow(n as u32 - 1);
        if k == midIndex {
            return '1';
        }
        if k < midIndex {
            return Self::find_kth_bit(n-1, k);
        }
        if Self::find_kth_bit(n-1, 2 * midIndex - k) == '0' {
            '1'
        } else {
            '0'
        }
    }
}

